docs/developer/payments/lifecycle.mdx

```diff
@@ -173,27 +258,126 @@ it may still remain unpaid if the pending amounts fail to process.
+
+#### Events deduplication
+
+Each event is identified by `pspReference`. This field is used to de-duplicate events in case there were reported multiple times by the payment app.
+
+For example if Saleor receives two `CHARGE_SUCCESS` events with the same `pspReference` and `amount` there will be only one event stored on Transaction. Second event will be treated as "already reported" and it won't be stored by Saleor.
+
+When reporting such events in `transactionEventReport` mutation it returns event that already exists and a flag `alreadyReported: true`.
+
+You can learn more about this mutation and event deduplication in [Reporting actions for transactions](/developer/payments/transactions.mdx#reporting-actions-for-transactions) docs.
+
+### State machine
+
+Saleor Transaction model **is not a state machine**. This allows great flexibility when integrating with many payment providers, even those that might implement some unconventional flows.
+However this also means that we do not forbid changes on Transactions that might be seen as invalid by your business use-case, such validation is on the Payment App side.
+
+For example:
+- Available `actions` on `TransactionItem` (list of `REFUND`, `CHARGE`, `CANCEL`) are honored by Saleor Dashboard, but are not rejected once `transactionRequestAction` is requested via API - such request is sent to payment app
+- Transaction might receive `CANCEL_SUCCESS` event even though it had `CHARGE_SUCCESS` reported (in such case to lower `chargedAmount` app would need to report `REFUND_SUCCESS` event).
+
+#### Existing validation
+
+With that in mind, Saleor prohibits following changes, that would otherwise cause incorrect internal calculations:
+- Multiple `AUTHORIZATION_SUCCESS` events with different either `pspReference` or `amount`
+  - Saleor will throw an exception: "Event with `AUTHORIZATION_SUCCESS` already reported for the transaction. Use `AUTHORIZATION_ADJUSTMENT` to change the authorization amount."
+
+:::note
+**Example**: `AUTHORIZATION_SUCCESS` event with `pspReference: psp1` and `amount: 10 USD`, then report `AUTHORIZATION_SUCCESS` event with `pspReference: psp2` and `amount: 10 USD`
+:::
+
+:::tip
+**Fix**: In order to change `authorizedAmount` value, payment app should send `AUTHORIZATION_ADJUSTMENT` event. If there are in fact two valid `AUTHORIZATION_SUCCESS` events, create new `TransactionItem`
+:::
+
+- Reporting the same event based on `type` and `pspReference` with different `amount`
+  - Saleor will throw an exception: "The transaction with provided `pspReference` and `type` already exists with different amount."
+
+:::note
+**Example**: `REFUND_SUCCESS` with `pspReference: psp1` and `amount: 10 USD`, then report `REFUND_SUCCESS` with `pspReference: psp1` and `amount: 21 USD`
+:::
+
+:::tip
+**Fix**: Previous event must be invalidated via `_FAILURE` event, then report `_SUCCESS` event again
+:::
```

Comment:
I'm not sure about this if this is in fact a fix - what should we do when we want to update event's amount with the same `pspReference`? 🤔

Reply:
We do not have EDIT option. Events are treated like history. If you want to modify the value, then you would need to work with existing event types. In your case, you will need to provide different pspReference i guess.

Reply:
OK, so the new `SUCCESS` event **must have** other `pspReference` otherwise it will be ignored by Saleor? @korycins

Reply:
I've just quickly checked this and nope, as you said it must have a different `pspReference` :/

```
Error returned when reporting event:
The transaction with provided `pspReference` and `type` already exists with different amount.
```

Reply:
But this is expected. You reported the charge action with PSP:123. This is the ID of this action. As you received it from payment-provider. Dont see a situation where payment-provider returns you a different value 🤔  At least this is what we expect when we were building the interface. 
Above you explain that events are like bank account history, which is really good example, but now you expect, that we will give you an option to modify the "history" 😅

Reply:
Yes, I know we have a history ledger and it's not modifiable - meaning that you cannot **edit** existing events, **but** we probably should have an option to make adjustments, that allow changing this history **by adding new events**.

Going back to the bank analogy: if they have some issue on their side, and you make a claim (e.g. they charged you more for some service than they should like ATM payout) they can make an correction for that charge by giving you money back. Same thing with invoices, if you made a mistake when issuing an invoice, you issue a correction detailing the changes.

Reply:
It's out of scope for this PR, here's a link for this feature idea: https://linear.app/saleor/issue/IFREQ-36/corrections-for-transactionevents